allensdk/brain_observatory/ecephys/stimulus_analysis.py

```diff
@@ -0,0 +1,227 @@
+from six import string_types
+import numpy as np
+import pandas as pd
+import scipy.stats as st
+import scipy.ndimage as ndi
+
+from .ecephys_session import EcephysSession
+
+
+class StimulusAnalysis(object):
+    def __init__(self, ecephys_session, **kwargs):
+        """
+        :param ecephys_session: an EcephySession object or path to ece nwb file.
+        """
+        if isinstance(ecephys_session, EcephysSession):
+            self._ecephys_session = ecephys_session
+        elif isinstance(ecephys_session, string_types):
+            nwb_version = kwargs.get('nwb_version', None)
+            self._ecephys_session = EcephysSession.from_nwb_path(path=ecephys_session, nwb_version=nwb_version)
+
+        self._cell_ids = None
+        self._cells_filter = kwargs.get('filter', None)  # {'location': 'probeC', 'structure_acronym': 'VISp'}
+        self._number_cells = None
+        self._spikes = None
+        self._stim_table = None
+        self._stim_table_spontaneous = None
+        self._stimulus_names = None
+        self._running_speed = None
+        self._sweep_events = None
+        self._mean_sweep_events = None
+        self._sweep_p_values = None
+        self._peak = None
+
+    @property
+    def ecephys_session(self):
+        return self._ecephys_session
+
+    @property
+    def cell_id(self):
```

Comment:
using cell vs. unit here is likely to be confusing. It would be nice if we had fully consistent terminology across projects, but:
1. consistent terminology within projects is required, while across projects is nice-to-have (and a harder question).
2. renaming units -> cells within ecephys has been explicitly rejected.